connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -1211,13 +1211,11 @@ class Column private[sql] (private[sql] val expr: proto.Expression) extends Logg
    * @group df_ops
    * @since 3.4.0
    */
-  def explain(extended: Boolean): Unit = {
+  def explain(extended: Boolean)(implicit spark: SparkSession): Unit = {
```

Comment:
Who will provide the implicit `SparkSession` in the users' context?

Reply:
Good question. It seems we must send the msg to server side. So we require `SparkSession`.
But it's a question how to get `SparkSession` in connect's `Column` ? or we define the SparkSession in `Column` directly?

Reply:
I am not sure what happened here, my comment seemed to have gone missing... Anyway, please don't send an RPC for this. Just use the local expression tree and create something that is a little bit easier to read.

Reply:
@hvanhovell If so, the output of `Column.explain` can't keep consistent with Spark SQL.

Reply:
@hvanhovell Do you mean to construct local catalyst expression tree at connect client side ?

Reply:
@beliefer no we can't be fully consistent with SQL. We are already deviating from what the original implementation does when you set `extended = true`. I just want something that is more readable.

As for the implementation. The catalyst dependency is going away. Just create a bit of tree traversal logic that nicely renders the expression tree.

Reply:
I understand what you mean roughly. We don't need follow SQL, but the output reflects the hierarchy of the tree.